### PR TITLE
Per-PR GKE smoke test.

### PR DESCRIPTION
**Pull request by spxtr**
Assigned to @ixdy, Jenkins oncall.

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 1]
```diff
@@ -165,6 +165,83 @@
...
+          skip-if-no-test-files: false  # We expect JUnit output from this job.
+          success-comment: ''
+          failure-comment: ''
+          error-comment: ''
+          trigger-phrase: 'e2e\s+test'
+          cmd: |
+            export KUBE_SKIP_PUSH_GCS=y
+            export KUBE_RUN_FROM_OUTPUT=y
+            export KUBE_FASTBUILD=true
+            ./hack/jenkins/build.sh
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: ugh.

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 2]
```diff
@@ -165,6 +165,83 @@
...
+             if [[ "${{ghprbTargetBranch:-}}" == "master" || "${{ghprbTargetBranch:-}}" == "release-1.2" || "${{ghprbTargetBranch:-}}" == "release-1.3" ]]; then
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: nit: looks like this is indented one space more than it should be?

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 3]
```diff
@@ -165,6 +165,83 @@
...
+              # Assume we're upping, testing, and downing a cluster
+              export E2E_UP="true"
+              export E2E_TEST="true"
+              export E2E_DOWN="true"
+
+              # Skip gcloud update checking
+              export CLOUDSDK_COMPONENT_MANAGER_DISABLE_UPDATE_CHECK=true
+
+              # AWS variables
+              export KUBE_AWS_INSTANCE_PREFIX=${{E2E_NAME}}
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: probably don't need this?

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 4]
```diff
@@ -165,6 +165,83 @@
...
+
+              # GCE variables
+              export INSTANCE_PREFIX=${{E2E_NAME}}
+              export KUBE_GCE_NETWORK=${{E2E_NAME}}
+              export KUBE_GCE_INSTANCE_PREFIX=${{E2E_NAME}}
+              export GCE_SERVICE_ACCOUNT=$(gcloud auth list 2> /dev/null | grep active | cut -f3 -d' ')
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: shouldn't need this either.

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 5]
```diff
@@ -165,6 +165,83 @@
...
+              export E2E_TEST="true"
+              export E2E_DOWN="true"
+
+              # Skip gcloud update checking
+              export CLOUDSDK_COMPONENT_MANAGER_DISABLE_UPDATE_CHECK=true
+
+              # AWS variables
+              export KUBE_AWS_INSTANCE_PREFIX=${{E2E_NAME}}
+
+              # GCE variables
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: probably don't need any of these either?

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 6]
```diff
@@ -165,6 +165,83 @@
...
+              export GCE_SERVICE_ACCOUNT=$(gcloud auth list 2> /dev/null | grep active | cut -f3 -d' ')
+
+              # GKE variables
+              export CLUSTER_NAME=${{E2E_NAME}}
+              export KUBE_GKE_NETWORK=${{E2E_NAME}}
+
+              # Get golang into our PATH so we can run e2e.go
+              export PATH=${{PATH}}:/usr/local/go/bin
+
+              timeout -k 15m 55m ./hack/jenkins/e2e-runner.sh && rc=$? || rc=$?
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: I kinda wish this wasn't so WET, but I guess we can maybe clean things up later.

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml (hunk 2 above)
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: Fixed.

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml (hunk 3 above)
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: Removed.

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml (hunk 5 above)
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: Removed.

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml (hunk 6 above)
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: Yeah, it's not beautiful but I'd rather copy-paste than make the YAML more confusing.

**Comment by ixdy** [comment]
LGTM. worst case is we fail, but we're not blocking merges yet, so that should be OK.

**Pull request merged**